### ISSUE TITLE
[clang-tidy] Labeled compound statement should be considered a compound statement

### ISSUE THREAD

**Issue by zygoloid** [llvm/llvm-project#194694]
At least the `readability-inconsistent-ifelse-braces` and `readability-braces-around-statements` checks consider `[[likely]] { ... }` to be a braced statement. But they do not consider `label: {{ ... }}` to be a braced statement. For example:

```c++
void g(), h();
void x(bool cond) {
  if (cond) {
    g();
  // No warning here.
  } else [[unlikely]] {
    h();
  }
}
void y(bool cond) {
  if (cond) {
    g();
  // Warning here.
  } else foo: {
    h();
  }
}
```

A labeled `else` block is a technique used in certain `CHECK` / `ASSERT` macros and similar constructs, and there doesn't seem to be any good reason to warn about this construct, especially given that other annotations in this position (notably, statement attributes) don't warn.

**Comment by localspook** [comment]
Should be a matter of replacing things like this:
https://github.com/llvm/llvm-project/blob/5249e5527f48e1dd6b13c3dabed4992b299bfcb9/clang-tools-extra/clang-tidy/readability/InconsistentIfElseBracesCheck.cpp#L21-L26
with `stripLabelLikeStatements`.

**Comment by horizenight** [comment]
I'd like to work on this. The current [`ignoreAttributed`](https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-tidy/readability/InconsistentIfElseBracesCheck.cpp) helper only unwraps `AttributedStmt`; replacing it with [`Stmt::stripLabelLikeStatements()`](https://github.com/llvm/llvm-project/blob/main/clang/include/clang/AST/Stmt.h) (as suggested above) would cover `LabelStmt` and `SwitchCase` too. Will also audit [`BracesAroundStatementsCheck.cpp`](https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-tidy/readability/BracesAroundStatementsCheck.cpp) for the same issue and add lit cases to [`inconsistent-ifelse-braces-attributes.cpp`](https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/test/clang-tidy/checkers/readability/inconsistent-ifelse-braces-attributes.cpp) covering the labeled-block form.

**Comment by localspook** [comment]
Go for it!

**Comment by gxyd** [comment]
Hey @horizenight , just checking if you are still intending to work on this issue? If you are, I'll leave this one and take a look at some other issue(s). Thank you 😄 .
